massenergize/api
# issue 484 super-admin nudge implementation

#### Summary 
Sends weekly report on activities like signups, actions done, actions placed in Todo of all communities to super-admins, and also, community activities to cadmins.

#### Testing Steps 

- Pull in the branch
- Run `make init` to install dependencies
The nudge is scheduled to trigger every `Monday at 00:00` to change that, locate `src/celery/app.py` file, in `app.conf.beat_schedule` dict, change the `schedule value` to `crontab (minute='*/10', hour='*') ` to trigger the nudge every 10mins. This change will blast all super-admins and cadmins with emails every 10mins 😂 . To alter the email list, locate `src/api/tasks.py` file and replace `all_community_admins` and `super_admins` variable in the `send_email` function call with either a `list of emails` or an `email`. 

- Now run `make celery` to start celery process with redis as a broker or `make celery-remote` to use amazon sqs
-  Open a new terminal and run `make beat` to start celery-beat process.
-  Wait patiently for 10mins 
- You should receive an email after 10mins.


#### Requirements 
```
- django-celery-beat
- pycurl==7.19.3.1 # THIS SPECIFIC VERSION

```
